### PR TITLE
[ADD] doc: section on Escaping vs Sanitizing

### PR DESCRIPTION
**Pull request by mart-e**
Source https://github.com/odoo/enterprise/pull/15297#issuecomment-768985663

### CODE REVIEW

File: doc/reference/security.rst [hunk 1]
```diff
@@ -257,6 +257,64 @@ While formatting the template differently would prevent such vulnerabilities.
         font-weight: bold;
     }
 
+Escaping vs Sanitizing
+----------------------
+
+**Escaping** converts `TEXT -> CODE`. It is absolutely mandatory to do it every
```
Comment (xmo-odoo) [inline doc/reference/security.rst:263]:
```suggestion
**Escaping** converts TEXT to CODE. It is absolutely mandatory to do it every
```

Comment (mart-e) [reply doc/reference/security.rst:263]: I was wondering about that. The change of style increases the importance. Could be bold but "Escaping" is already in bold...

File: doc/reference/security.rst [hunk 2]
```diff
@@ -257,6 +257,64 @@ While formatting the template differently would prevent such vulnerabilities.
...
+
+    # Escaping turns it into CODE, good!
+    >>> code = html_escape(data)
+    >>> code
+    '&lt;R&amp;D&gt;'
+
+    # Now you can mix it with other code...
+    >>> self.message_post(body="<strong>%s</strong>" % code)
+
+**Sanitizing** converts `CODE -> SAFER_CODE`. It does not work on `TEXT`.
```
Comment (xmo-odoo) [inline doc/reference/security.rst:287]: Might be worth emphasizing that SAFER_CODE does not necessarily mean *safe* code, sanitization / validation can have holes and edge cases, which usually is not the case for escaping (this is kinda-sorta on the edge of [parse don't validate](https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/)). So in the spectrum between safe and unsafe, sanitisation is way closer to "unsafe" than escaping.

Comment (xmo-odoo) [reply doc/reference/security.rst:287]: (and sanitisation safety is balanced against sanitisation breakages: the safer the sanitisation the more likely it is to break things)

File: doc/reference/security.rst [hunk 3]
```diff
@@ -257,6 +257,72 @@ While formatting the template differently would prevent such vulnerabilities.
...
+the data comes from, and the desired features. The sanitization safety is
+balanced against sanitization breakages: the safer the sanitisation the more
+likely it is to break things.
+
+.. code-block:: python
+
+    >> code = "<div class='text-warning'>Important Information</div>"
+    # this will remove the style, which may break features
+    # but is necessary if the source is untrusted
+    >> html_sanitize(code, strip_classes=True)
```
Comment (odony) [inline doc/reference/security.rst:324]: including the result of this evaluation would be helpful, but it demoes a very weird behavior of html_sanitize:

```py
In [3]: odoo.tools.html_sanitize(code, strip_classes=True)
Out[3]: 'Important Information'
```

I guess you could just use a `<p>` for the sake of having a simple, non-surprising example:
```py
In [4]: code = "<p class='text-warning'>Important Information</p>"

In [5]: odoo.tools.html_sanitize(code, strip_classes=True)
Out[5]: '<p>Important Information</p>'
```

**Comment by mart-e** [comment]
@robodoo r+

**Pull request closed without merging**